test/e2e/network/network_policy.go

```diff
@@ -532,12 +640,601 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+				testCanConnect(f, f.Namespace, "client-a", serviceB, 80)
+			})
+			// client-a should be able to contact server as the policy is to allow all
+			ginkgo.By("Creating client-a which should be able to contact the server-a.", func() {
+				testCanConnect(f, f.Namespace, "client-a", service, 80)
+			})
+		})
+
+		ginkgo.It("should stop enforcing policies after they are deleted [Feature:NetworkPolicy]", func() {
+			ginkgo.By("Creating a network policy for the server which denies all traffic.")
+			policyDenyAll := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "deny-all",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Deny all traffic
+					PodSelector: metav1.LabelSelector{},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					Ingress:     []networkingv1.NetworkPolicyIngressRule{},
+				},
+			}
+
+			policyDenyAll, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyDenyAll)
+			framework.ExpectNoError(err, "Error occurred while creating policy: policyDenyAll.")
+			defer cleanupNetworkPolicy(f, policyDenyAll)
+
+			ginkgo.By("Creating client-a which should not be able to contact the server.", func() {
+				testCannotConnect(f, f.Namespace, "client-a", service, 80)
+			})
+
+			ginkgo.By("Creating a network policy for the server which allows traffic only from client-a.")
+			policyAllowFromClientA := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "allow-from-client-a-pod-selector",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply this policy to the Server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-name": podServerLabelSelector,
+						},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					// Allow traffic from "client-a"
+					Ingress: []networkingv1.NetworkPolicyIngressRule{{
+						From: []networkingv1.NetworkPolicyPeer{{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"pod-name": "client-a",
+								},
+							},
+						}},
+					}},
+				},
+			}
+
+			policyAllowFromClientA, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowFromClientA)
+			framework.ExpectNoError(err, "Error occurred while creating policy: policyAllowFromClientA.")
+
+			ginkgo.By("Creating client-a which should be able to contact the server.", func() {
+				testCanConnect(f, f.Namespace, "client-a", service, 80)
+			})
+
+			ginkgo.By("Deleting the network policy allowing traffic from client-a")
+			cleanupNetworkPolicy(f, policyAllowFromClientA)
+
+			ginkgo.By("Creating client-a which should not be able to contact the server.", func() {
+				testCannotConnect(f, f.Namespace, "client-a", service, 80)
+			})
```

Comment:
For completeness, delete `policyDenyAll` now and confirm that client-a *can* connect again.

Reply:
Added the test after deleting the `policyDenyAll `